### PR TITLE
View Variables remove component button and functionality.

### PR DESCRIPTION
**Pull request by Gerhazo**
## About the PR 
Adds a "remove component" button to view variables to remove components present on target, to complement the already existing "add component" one.

## Why's this needed? 
Allows removing components in a convenient manner to supplement the currently present ability to add them.

### CODE REVIEW

File: code/modules/admin/debug.dm [hunk 1]
```diff
@@ -1346,3 +1346,21 @@ var/datum/flock/testflock
...
 
 	boutput(usr, "<span class='notice'>Returned: [!isnull(returnval) ? returnval : "null"]</span>")
+
+/proc/debugRemoveComponent(var/datum/target = null)
+	var/list/dc = target.datum_components
+	if(!dc)
+		boutput(usr, "<span class='notice'>No components present on [target].</span>")
+		return
+
+	var/list/comps = dc[/datum/component]
```
Comment (ZeWaka) [inline code/modules/admin/debug.dm:1356]: indexing... by type?

File: code/modules/admin/debug.dm [hunk 2]
```diff
@@ -1346,3 +1346,21 @@ var/datum/flock/testflock
 
 
 	boutput(usr, "<span class='notice'>Returned: [!isnull(returnval) ? returnval : "null"]</span>")
+
+/proc/debugRemoveComponent(var/datum/target = null)
+	var/list/dc = target.datum_components
```
Comment (ZeWaka) [inline code/modules/admin/debug.dm:1351]: untyped list?

File: code/modules/admin/debug.dm (hunk 1 above)
Comment (Gerhazo) [reply code/modules/admin/debug.dm:1356]: It sounds weird, but components are stored in a somewhat peculiar way.

For reference, for the sake of consistency I tried to be close to components' TransferComponents proc, which iterates through them in a similar way: https://github.com/goonstation/goonstation/blob/2e116d0e8df159dcd84e5cc7625d111b5223eaaf/code/datums/components/_component.dm#L541

File: code/modules/admin/debug.dm (hunk 2 above)
Comment (Gerhazo) [reply code/modules/admin/debug.dm:1351]: datum_components is present on every datum and it is a regular untyped list.

**Review by ZeWaka — APPROVED** [review]

**Pull request merged**